### PR TITLE
Update markdown rendering

### PR DESCRIPTION
**Pull request by beyang**
Continuation of https://github.com/sourcegraph/sourcegraph/pull/46361

## Test plan

Tested locally and updated unit test

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.51% (+14.35 kb) 🔺 | 0.13% (+19.25 kb) 🔺 | 0.04% (+4.89 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/bl/SuperAuguste/update-markdown/compare-report.html;tab=live_object) for a full comparison between the commits 95dee3f and 9449278 or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

File: cmd/frontend/internal/markdown/markdown.go [hunk 1]
```diff
@@ -23,9 +32,41 @@ func Render(content string) string {
...
+						chromahtml.WithClasses(true),
+						chromahtml.WithLineNumbers(false),
+					),
+				),
+			),
+			goldmark.WithParserOptions(
+				parser.WithAutoHeadingID(),
+			),
+			goldmark.WithRendererOptions(
+				html.WithUnsafe(),
```
Comment (camdencheek) [inline cmd/frontend/internal/markdown/markdown.go:62]: nit: I think it's worth adding a note why the unsafe option is okay here

```suggestion
				// HTML sanitization is handled by bluemonday
				html.WithUnsafe(),
```

File: cmd/frontend/internal/markdown/markdown.go [hunk 2]
```diff
@@ -23,9 +32,41 @@ func Render(content string) string {
...
+		)
 	})
 
-	unsafeHTML := gfm.Markdown([]byte(content))
-	return string(policy.SanitizeBytes(unsafeHTML))
+	var buf bytes.Buffer
+	if err := renderer.Convert([]byte(content), &buf); err != nil {
+		return "", err
+	}
+	return string(policy.SanitizeBytes(buf.Bytes())), nil
```
Comment (camdencheek) [inline cmd/frontend/internal/markdown/markdown.go:71]: nit: we can skip some string/bytes conversions by using `strings.Builder`

```suggestion
	var buf strings.Builder
	if err := renderer.Convert([]byte(content), &buf); err != nil {
		return "", err
	}
	return policy.Sanitize(buf.String()), nil
```

**Review by camdencheek — APPROVED** [review]
This looks great! I did a little testing locally, and things generally rendered more consistently. Just a couple of nits inline.

**Pull request merged**